connector/profilingmetricsconnector/connector.go

```diff
@@ -0,0 +1,174 @@
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package profilingmetricsconnector // import "github.com/elastic/opentelemetry-collector-components/connector/profilingmetricsconnector"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+
+	semconv "go.opentelemetry.io/otel/semconv/v1.34.0"
+)
+
+// profilesToMetricsConnector implements xconnector.Profiles
+type profilesToMetricsConnector struct {
+	nextConsumer consumer.Metrics
+	config       *Config
+}
+
+// Capabilities returns the consumer capabilities.
+func (c *profilesToMetricsConnector) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{
+		MutatesData: false,
+	}
+}
+
+// ConsumeProfiles processes profiles data and extracts metrics.
+func (c *profilesToMetricsConnector) ConsumeProfiles(ctx context.Context, profiles pprofile.Profiles) error {
+	metrics := c.extractMetricsFromProfiles(profiles)
+
+	if metrics.MetricCount() > 0 {
+		return c.nextConsumer.ConsumeMetrics(ctx, metrics)
+	}
+
+	return nil
+}
+
+func (c *profilesToMetricsConnector) Start(ctx context.Context, host component.Host) error {
+	return nil
+}
+
+func (c *profilesToMetricsConnector) Shutdown(ctx context.Context) error {
+	return nil
+}
+
+// extractMetricsFromProfiles extracts basic metrics from the profiles data.
+func (c *profilesToMetricsConnector) extractMetricsFromProfiles(profiles pprofile.Profiles) pmetric.Metrics {
+	metrics := pmetric.NewMetrics()
+
+	dictionary := profiles.ProfilesDictionary()
+	resourceProfiles := profiles.ResourceProfiles()
+	for i := 0; i < resourceProfiles.Len(); i++ {
+		resourceProfile := resourceProfiles.At(i)
+		resourceMetrics := metrics.ResourceMetrics().AppendEmpty()
+
+		// Copy resource attributes
+		resourceProfile.Resource().Attributes().CopyTo(resourceMetrics.Resource().Attributes())
+
+		// Process each scope's profiles
+		scopeProfiles := resourceProfile.ScopeProfiles()
+		for j := 0; j < scopeProfiles.Len(); j++ {
+			scopeProfile := scopeProfiles.At(j)
+			scopeMetrics := resourceMetrics.ScopeMetrics().AppendEmpty()
+
+			// Copy scope information
```

Comment:
Do we want to also include off-cpu here? Maybe it's best to leave off-cpu out initially to avoid introducing noise? Also, I don't think there's away to currently distinguish between on-cpu/off-cpu just based on scope (only based on unit), we should think about adding that.

Reply:
At the moment, `scopeMetrics.Scope()` comes down to this:
```
  "scope.name": [
    "otelcol-ebpf-profiler"
  ]
```
So, as you mentioned, it is not possible atm to distinguish between different kinds of profiling from this scope metric.
Do you prefer to limit the metrics generation to on-cpu only to would you like to see to extend the scope metrics, to include the profiling kind information, extracted from profile information?